src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,38 @@
+interface IMetadata {
+  kind: MetadataKind
+  version: string
+}
+
+export enum MetadataKind {
+  REFERRAL = 'referrer',
+}
+
+export type Metadata = {
+  [MetadataKind.REFERRAL]: IReferralMetadata
+}
+
+export type MetadataDoc = {
```

Comment:
Now that I look at it, there are too many metadata types.
A better name for this could be `AppDataDoc`.